apps/functions-subscription/src/domain/activation.ts

```diff
@@ -0,0 +1,59 @@
+import * as TE from 'fp-ts/TaskEither';
+import * as t from 'io-ts';
+import { IsoDateFromString } from '@pagopa/ts-commons/lib/dates';
+import { NonNegativeNumber } from '@pagopa/ts-commons/lib/numbers';
+import { TrialIdCodec, UserIdCodec } from './subscription';
+
+const BaseActivationItemCodec = t.strict({
+  id: t.string,
+  trialId: TrialIdCodec,
+  _etag: t.string,
+});
+export type BaseActivationItemCodec = t.TypeOf<typeof BaseActivationItemCodec>;
+
+export const ActivationRequestItemCodec = t.intersection([
+  BaseActivationItemCodec,
+  t.strict({
+    userId: UserIdCodec,
+    createdAt: IsoDateFromString,
+    type: t.literal('request'),
+    activated: t.boolean,
+  }),
+]);
+export type ActivationRequestItem = t.TypeOf<typeof ActivationRequestItemCodec>;
+
+export const ActivationJobItemCodec = t.intersection([
+  BaseActivationItemCodec,
+  t.strict({
+    createdAt: IsoDateFromString,
+    usersToActivate: NonNegativeNumber,
+    usersActivated: NonNegativeNumber,
+    type: t.literal('job'),
+  }),
+]);
+export type ActivationJobItem = t.TypeOf<typeof ActivationJobItemCodec>;
+
+export type ActivationResult = 'success' | 'fail' | 'not-executed';
```

Comment:
**question**: When do you think we need the `not-executed` status? I'm asking because maybe we can merge it with the `success` status.